### PR TITLE
settings: Implement the contrast settings

### PR DESCRIPTION
**Pull request by hfiguiere**
See https://github.com/flatpak/xdg-desktop-portal/pull/1175

Note: I still have questions how this work on a non GNOME system.... so it might not be relevant.

Mostly a cut and paste of https://gitlab.gnome.org/GNOME/xdg-desktop-portal-gnome/-/merge_requests/143

### CODE REVIEW

**Comment by TingPing** [comment]
> Note: I still have questions how this work on a non GNOME system.... so it might not be relevant.

This is the only setting GTK looks at to detect high contrast. Other desktops have to set the GSetting. KDE already sets various GSettings for theme, not sure if they set this one too but that would be changed there.

**Review by TingPing — APPROVED** [review]

**Comment by dominichayesferen** [comment]
> > Note: I still have questions how this work on a non GNOME system.... so it might not be relevant.
> 
> This is the only setting GTK looks at to detect high contrast. Other desktops have to set the GSetting. KDE already sets various GSettings for theme, not sure if they set this one too but that would be changed there.

For the record, last I checked they don't... because they don't have High Contrast in that regard yet.

File: src/settings.c [hunk 1]
```diff
@@ -102,6 +104,19 @@ get_color_scheme (void)
   return g_variant_new_uint32 (color_scheme);
 }
 
+static GVariant *
+get_contrast_value ()
+{
+  SettingsBundle *bundle = g_hash_table_lookup (settings, "org.gnome.desktop.a11y.interface");
+  g_autofree char *theme = NULL;
```
Comment (TingPing) [inline src/settings.c:111]: theme is unused

Comment (hfiguiere) [reply src/settings.c:111]: and the compiler doesn't even throw a warning...

Comment (TingPing) [reply src/settings.c]: It is the one downside to the `g_auto*` usage, its always used because its usually assigned to `NULL`.

**Pull request merged**